infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -116,24 +122,30 @@ private void openTables() {
         }
     }
 
-    public Set<String> getStreamsToReplicate() {
+    /**
+     * This method is for ACTIVE to fetch streams to replicate from LOG_REPLICATION_STREAMS_INFO_TABLE.
+     * If it doesn't exist, we will create info table by reading from registry table.
+     *
+     * @return Set of TableInfo for streams to replicate from ACTIVE
+     */
+    public Set<TableInfo> fetchStreamsToReplicate() {
         try {
             currentVersion = logReplicationConfigAdapter.getVersion();
-            Set<String> fetchedStreams = logReplicationConfigAdapter.fetchStreamsToReplicate();
+            Set<TableInfo> fetchedStreams = readStreamsToReplicateFromRegistry(this.runtime.getAddressSpaceView().getLogTail());
 
             return IRetry.build(IntervalRetry.class, () -> {
-                Set<String> streams = new HashSet<>();
+                Set<TableInfo> streamsInfo = new HashSet<>();
                 try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
                     VersionResult result = verifyVersionResult(txn);
                     if (result.equals(VersionResult.UNSET) || result.equals(VersionResult.CHANGE)) {
                         boolean isUpgraded = result.equals(VersionResult.CHANGE);
                         // Case of upgrade or initial boot: clear table entries and sync with plugin info
                         clearTables();
                         updateVersionTable(txn, isUpgraded);
-                        streams = updateStreamsTable(txn, fetchedStreams);
+                        streamsInfo = updateStreamsTable(txn, fetchedStreams);
                     }
                     txn.commit();
-                    return streams.isEmpty() ? readStreamsToReplicateFromTable() : streams;
+                    return streamsInfo.isEmpty() ? readStreamsToReplicateFromTable() : streamsInfo;
```

Comment:
This goes along with the comment on L131, readStreamsToReplicateFromTable() is called only from LogReplicationConfig.addStreams(). I couldn't find any other usage. 
We need not write the newly found streams on standby to the table since we anyways recalculate streamsToReplicate when we see a new stream.

I suppose this readStreamsToReplicateFromTable() is used when reading streamsToReplicate from static file?

Reply:
Also, if the intentions to retain readStreamsToReplicateFromTable was to support the static file, can we check if its possible to separate out the logic for consuming from static file and dynamically calculating streamsToReplicate?